airflow/models/baseoperator.py

```diff
@@ -767,7 +767,13 @@ def __init__(
         self.email = email
         self.email_on_retry = email_on_retry
         self.email_on_failure = email_on_failure
+
+        if execution_timeout is not None and not isinstance(execution_timeout, timedelta):
```

Comment:
```suggestion
        if not isinstance(execution_timeout, timedelta):
```
should be enough as we already have the default set in ctor:

https://github.com/apache/airflow/blob/d4d29718fdc46cd49e8b5be2e54fb313d595f06e/airflow/models/baseoperator.py#L205

Reply:
Even if you have plan to allow None, the following should work:

```
if execution_timeout and not isinstance(execution_timeout, timedelta):
```

Reply:
nice, updated.